File: src/ts/magma/app/compile/compose/StripComposable.ts
import { Composable } from "../../../../magma/app/compile/compose/Composable";
import { Option } from "../../../../magma/api/option/Option";
export class StripComposable<T> implements Composable<string, T> {
	composable: Composable<string, T>;
	constructor (composable: Composable<string, T>) {
		this.composable = composable;
	}
	apply(s: string): Option<T> {
		return this.composable.apply(s.strip())/*unknown*/;
	}
}

File: src/java/magma/app/compile/compose/StripComposable.java
package magma.app.compile.compose;

import magma.api.option.Option;

public record StripComposable<T>(Composable<String, T> composable) implements Composable<String, T> {
    @Override
    public Option<T> apply(String s) {
        return this.composable.apply(s.strip());
    }
}
